homeassistant/components/lg_thinq/mqtt.py

```diff
@@ -0,0 +1,165 @@
+"""Support for LG ThinQ Connect API."""
+
+from __future__ import annotations
+
+import asyncio
+from datetime import datetime
+import json
+import logging
+from typing import Any
+
+from thinqconnect import ThinQApi, ThinQMQTTClient
+
+from homeassistant.core import Event, HomeAssistant
+
+from .const import DEVICE_PUSH_MESSAGE, DEVICE_STATUS_MESSAGE
+from .coordinator import DeviceDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ThinQMQTT:
+    """A class that implements MQTT connection."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        thinq_api: ThinQApi,
+        client_id: str,
+        coordinator_map: dict[str, DeviceDataUpdateCoordinator],
+    ) -> None:
+        """Initialize a mqtt."""
+        self.hass = hass
+        self.thinq_api = thinq_api
+        self.client_id = client_id
+        self.coordinator_map = coordinator_map
+        self.client: ThinQMQTTClient | None = None
+
+    async def async_connect(self) -> bool:
+        """Create a mqtt client and then try to connect."""
+        try:
+            self.client = await ThinQMQTTClient(
+                self.thinq_api, self.client_id, self.on_message_received
+            )
+            if self.client is None:
+                return False
+
+            # Connect to server and create certificate.
+            return await self.client.async_prepare_mqtt()
+        except Exception as exc:  # noqa: BLE001
+            _LOGGER.error("Failed to connect: %s", exc)
+            return False
+
+    async def async_disconnect(self, event: Event | None = None) -> None:
+        """Unregister client and disconnects handlers."""
+        await self.async_end_subscribes()
+
+        if self.client is not None:
+            try:
+                await self.client.async_disconnect()
+            except Exception as exc:  # noqa: BLE001
+                _LOGGER.error("Failed to disconnect: %s", exc)
```

Comment:
```suggestion
                _LOGGER.exception("Failed to disconnect: %s", exc)
```
This will log the stacktrace as well